server/src/controllers/workspace.js

```diff
@@ -0,0 +1,249 @@
+import Logger from '../logger';
+import Location from '../models/location';
+import Workspace from '../models/workspace';
+
+const logger = Logger('controller > workspace');
+
+exports.getWorkspaces = async (request, response, next) => {
+  const { qLocation, page = 1, count = 10 } = request.params;
+  try {
+    if (qLocation === undefined) {
+      logger.error('Search Value is empty');
+      response.status(404).json({
+        error: 'Empty query in the search..',
+      });
+      return next();
+    }
+    logger.info(`Exploring workspace at location: ${qLocation}`);
+    const locationInDB = await Location.findOne({ location: qLocation });
+    if (locationInDB === null) {
+      response.status(404).json({
+        error: `No workspaces in ${qLocation}`,
+      });
+      return next();
+    }
+    const arrayOfWorkspaces = [...locationInDB.workspaces];
+    const startIndex = (parseInt(page, 10) - 1) * parseInt(count, 10);
+    const endIndex = (parseInt(page, 10) - 1) * parseInt(count, 10) + parseInt(count, 10);
+    response.status(200).json({
+      status: 'success',
+      workspaces: arrayOfWorkspaces.slice(startIndex, endIndex),
+    });
+  } catch (error) {
+    logger.error(error);
+    response.status(500);
+  }
+  return next();
+};
+
+exports.addWorkspaceImages = async (request, response, next) => {
```

Comment:
function too complex

Reply:
Distributed chunks of operation in schema statics functions